spec/system/posts_index_integration_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts Index Page', type: :system do
+  before do
+    @user = create(:user)
+    @posts = create_list(:post, 5, author: @user)
+    @posts.each { |post| create_list(:comment, 3, post:, author: @user) }
+
+    visit user_posts_path(@user)
+  end
+
+  it 'displays the user\'s profile picture' do
+    expect(page).to have_selector("img[src='#{@user.photo}']")
+  end
+
+  it 'displays the user\'s name' do
+    expect(page).to have_link(@user.name, href: user_path(@user))
+  end
+
+  it 'displays the number of posts the user has written' do
+    expect(page).to have_content("Number of posts: #{@user.posts_counter}")
+  end
+
+  it 'displays each post\'s title' do
+    @posts.each do |post|
+      expect(page).to have_content(post.title)
+    end
+  end
+
+  it 'displays each post\'s text (body)' do
+    @posts.each do |post|
+      expect(page).to have_content(post.text)
+    end
+  end
+
+  it 'displays each post\'s five most recent comments' do
+    @posts.each do |post|
+      post.comments.each do |comment|
+        expect(page).to have_content(comment.text)
+      end
+    end
+  end
```

Comment:
- [ ]  @Team Great work adding tests for the posts' index page, however on this test, you are supposed to check if the five_most_recent_comments method is working correctly, that given more than five comments it will display five recent comments